syntax = "proto3";
package auth;

import "google/protobuf/timestamp.proto";
import "google/protobuf/empty.proto";

import "paperpilot_common/protobuf/user/user.proto";


// 登录请求
message LoginRequest {
    // 手机号
    string phone = 1;
    // 密码
    string password = 2;
}

// token
message Token {
    // token值
    string value = 1;
    // 过期时间
    google.protobuf.Timestamp expire_time = 2;
}

// 登录响应
message LoginResponse {
    // 用户信息
    user.UserInfo user = 1;
    // access token
    Token access = 2;
    // refresh token
    Token refresh = 3;
}

// 刷新token请求
message RefreshTokenRequest {
    // refresh token
    string refresh = 1;
}

// 刷新token响应
message RefreshTokenResponse {
    // access token
    Token access = 1;
}


// 认证公开接口（可匿名访问）
service AuthService {
    // 登录
    rpc Login(LoginRequest) returns (LoginResponse);

    // 刷新token
    rpc Refresh(RefreshTokenRequest) returns (RefreshTokenResponse);

    // 登出
    rpc Logout(google.protobuf.Empty) returns (google.protobuf.Empty);

    // 注册
    rpc Register(user.CreateUserRequest) returns (user.UserDetail);
}
